### PR TITLE
MacOS CI Workflow File

### PR DESCRIPTION
**Pull request by s0600204**
This PR provides a workflow for a MacOS CI. For now it must be manually activated, however [here is a recent run](https://github.com/s0600204/Vega-Strike-Engine-Source/actions/runs/929734201) to give some idea of what the end result is.

As mentioned in the related issue (#78), both `OpenAL` and `OpenGL` (or at least the Apple-provided implementations of them) are now considered deprecated on MacOS. Thus the workflow file, along with testing building with both `clang` and `gcc`, also provide jobs that attempt to use alternate `OpenAL` and `OpenGL` implementations (`OpenAL-soft` and `Mesa3D`/`freeGLUT` respectively).

The PR does not contain any attempted fixes for building on MacOS, other than that needed to facilitate using `OpenAL-soft`, `Mesa3D`, and `freeGLUT`. (For those, see the below "related branch".)

----

Code Changes:
- [x] This is a CI-Change, no gameplay modifications made

Related Issues:
- #78
- #170 (Discussion of moving away from OpenGL)

Related branch:
- [WIP MacOS Build Fixes](https://github.com/s0600204/Vega-Strike-Engine-Source/tree/macosx)
    - (Feel free to adopt this branch upstream as proceeding any further is somewhat beyond my capabilities at this time.)

### CODE REVIEW

**Comment by BenjamenMeyer** [comment]
@stephengtuggy can you take a look at this. Looks good to me but want a second opinion before approving/merging

File: .github/workflows/macos-ci.yml
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:1]: Please remove the spaces in the name. I find that these CI workflows work best when their internal name matches their filename.

File: .github/workflows/macos-ci.yml
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:5]: Looks good

File: .github/workflows/macos-ci.yml
Comment (stephengtuggy) [inline .github/workflows/macos-ci.yml:10]: Might be better to specify `macos-10.15` specifically, but we can do that later. https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

File: engine/CMakeLists.txt
Comment (stephengtuggy) [inline engine/CMakeLists.txt:934]: Hmm. Okay

File: script/cibuild
Comment (stephengtuggy) [inline script/cibuild:35]: 👍🏽

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
Looks good. The only changes I would like to see are minor ones. Specifically, the workflow name (on the very first line) is the only thing I would block on. Simply because it makes things awkward if we change it later.

**Comment by stephengtuggy** [comment]
Thanks for the PR, @s0600204 ! Glad to see you active on the project again.

**Review by stephengtuggy — APPROVED** [review]
Looks good to me!

**Pull request merged**
